### DISCUSSION TITLE
startup time increase in v0.13

### DISCUSSION THREAD

**Discussion by Quicksilver151** [neovide/neovide#2664] [Q&A]
Why is there such a large increase of startup time between version 0.12 and 0.13?

![Screenshot_20240620_213300-1](https://github.com/neovide/neovide/assets/56493793/a655ae7a-204b-429b-bbc9-7afe0537fab4)

**Comment by fredizzimo** [comment]
I'm not sure what this actually measures, since `neovide --no-fork` should never exit, and if it does, it measures both the startup and shutdown time. `neovide --fork` on the other hand just measures how long the initial process, which launches the real one takes to launch and exit. Which again does not really measure startup time, since it also measures how long it takes to exit, and ignores the time it takes for the actual process to load and show the window.

A more accurate way to measure this would be a profiler, or even a video capture and meassure the time from that.

But there are a couple of differences between 0.12 and 0.13
1. 0.12 does not wait for Neovim to properly exit before closing, and can even leave a ghost process behind due to that.
2. 0.13, doesn't show the window until Neovim tries to draw something. 0.12, shows the window earlier, but resizes it later, when the actual size is known. This could lead to visual glitches especially if your compositor animates the initial Windows.

**Reply by Quicksilver151** [reply]
--no-fork is not used. it still spawns a new instance  every time... or atleast it used to. strangely i cant seem to reproduce that behavior. --fork now seems to be a requirement now to launch as a separate process. very odd.

also, there is significant observable lag in opening an instance of neovide. i though getting time was better than screen recording. so i presented that. i suppose the time measured in the above post wildly innacurate. the difference is smaller. but still does exist.

>0.12 does not wait for Neovim to properly exit before closing, and can even leave a ghost process behind due to that

also, time for closing is not relevant as this is just about startup time. i mainly used neovide because it was faster than my terminal emulator to make a new instance. a little more startup time isnt that bad but i guess it could be worse.

>0.13, doesn't show the window until Neovim tries to draw something. 0.12, shows the window earlier, but resizes it later, when the actual size is known.

both show screen and resize at the same speed after it is launched. the delay is between when the the command is run, and when the window is drawn


btw...... do i submit a different discusssion for why the fonts now have a gap in neovide 0.13 (left). instead of how things are in neovide 0.12 (right). if it helps, running neovim in any termianl looks like the image on the right

![Screenshot_20240621_145918](https://github.com/neovide/neovide/assets/56493793/fc48f93b-c583-461b-8646-41266184b897)
